server/src/com/google/refine/ValidateHostHandler.java

```diff
@@ -84,16 +90,20 @@ public boolean isValidHost(String host) {
     }
 
     @Override
-    public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
-            throws IOException, ServletException {
-        String host = request.getHeader("Host");
+    public boolean handle(Request request, Response response, Callback callback)
+            throws Exception {
+        String host = request.getHttpURI().getHost();
         if (isValidHost(host)) {
-            super.handle(target, baseRequest, request, response);
+            return super.handle(request, response, callback);
         } else {
             // Return HTTP 404 Not Found, since we are
             // not serving content for the requested URL
-            response.sendError(HttpServletResponse.SC_NOT_FOUND, "Invalid hostname");
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            // TODO: Do we want to return JSON here?
```

Comment:
I think JSON might be best so it's more universal for all platforms/extensions/clients? dunno.
@wetneb any opinions?